tools/clang/lib/SPIRV/TypeTranslator.cpp

```diff
@@ -620,8 +620,12 @@ TypeTranslator::getLayoutDecorations(const DeclContext *decl, LayoutRule rule) {
 }
 
 uint32_t TypeTranslator::translateResourceType(QualType type, LayoutRule rule) {
+  // Resource types are either represented like C struct or C++ class in the
+  // AST. Samplers are represented like C struct, so isStructureType() will
+  // return true for it; textures are represented like C++ class, so
+  // isClassType() will return true for it.
+
   const auto *recordType = type->getAs<RecordType>();
-  assert(recordType);
```

Comment:
why are you removing this assert? the `recordType` pointer is used below and it's better to hit an assert than hit a segfault IMHO.

Reply:
Accidentally removed for debugging previously. Added back.